angular-1/src/app/app.component.ts

```diff
@@ -43,19 +44,28 @@ export class AppComponent implements OnInit, OnDestroy {
     // 2. Since we don't want to spam our service add filter by input value and do not call API until a user enters at least 3 chars.
 
     // 3. Add debounce to prevent API calls until user stop typing.
-
-    this.charactersResults$ = this.searchTermByCharacters
-      .pipe
+    this.charactersResults$ = this.searchTermByCharacters.pipe(
       // YOUR CODE STARTS HERE
-
-      // YOUR CODE ENDS HERE
-      ();
+      filter((value) => value.length >= 3),
+      debounceTime(1500),
+      switchMap((value) => this.mockDataService.getCharacters(value)),
+      takeUntil(this.destroy$)
```

Comment:
If I'm not mistaken, you should unsubscribe manually using `takeUntil()` only in case of a subsequent manual subscription using `subscribe()`.